spec/entities/base_spec.rb

```diff
@@ -51,37 +59,103 @@
   end
 
   describe 'request' do
-    let(:fixture) { File.read("spec/savon/fixtures/project/does_not_exists.xml") }
-    let(:client) { mock('client') }
-    let(:response) { mock('response') }
-    let(:request) { mock('request') }
-    let(:options) { mock('options') }
+    context 'with successful responses' do
+      let(:exists_response) { {'exists' => false} }
+      let(:bases_response) { {'bases' => {'id' => 1}} }
+      let(:prefix_bases_response) { {'bases' => {'id' => 2}} }
+
+      let(:connection) { Faraday.new { |builder| builder.adapter :test do |stubs|
+          stubs.get('/bases/1/exists') { |env| [200, {}, exists_response] }
+          stubs.get('/bases/') { |env| [200, {}, bases_response] }
+          stubs.get('/prefix/bases/') { |env| [200, {}, prefix_bases_response] }
+        end
+      }}
 
-    before :each do
-      options.expects(:timeout=)
-      options.expects(:open_timeout=)
-      request.expects(:url).with('/bases/exists')
-      request.expects(:body=).with({id: 1})
-      request.expects(:options).twice.returns(options)
+      before :each do
+        subject.class.stubs(:client).returns(connection)
+        subject.class.stubs(:endpoint).returns('bases')
+      end
+
+      context 'without an id parameter' do
+        context 'without a prefix' do
+          it 'should make the request without the prefix' do
+            response = subject.class.request('', {}, :get)
+            expect(response).to eq(bases_response)
+          end
+        end
+
+        context 'with a prefix' do
+          it 'should make the request with the prefix' do
+            response = subject.class.request('', {}, :get, 'prefix')
+            expect(response).to eq(prefix_bases_response)
+          end
+        end
+      end
+
+      context 'with an id parameter' do
+        it 'should make the request with the id included' do
+          response = subject.class.request(':id/exists', {id: 1}, :get)
+          expect(response).to eq(exists_response)
+        end
+      end
     end
 
-    context 'for the KalibroClient::Entities module' do
-      it 'should successfully get the Kalibro version' do
-        response.expects(:body).returns({exists: false})
-        client.expects(:post).yields(request).returns(response)
-        KalibroClient::Entities::Base.expects(:client).with(any_parameters).returns(client)
-        expect(KalibroClient::Entities::Base.request('exists', {id: 1})[:exists]).to eq(false)
+    context 'when the record was not found' do
+      context 'and or the status is 404' do
+        let!(:faraday_stubs) {
+          Faraday::Adapter::Test::Stubs.new do |stub|
```

Comment:
There's a missing call to `stubs.verify_stubbed_calls` to check wheter the Farady stubs were actually called.